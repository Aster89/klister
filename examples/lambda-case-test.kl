#lang "prelude.kl"

(import "lambda-case.kl")
(import "list.kl")

(define classify
  (lambda-case
    [(nil) 'nil]
    [(:: a b) 'cons]))

(example (classify (nil)))
(example (classify (:: 1 (:: 2 (nil)))))
