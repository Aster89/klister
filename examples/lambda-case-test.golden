#[lambda-case-test.kl:8.13-8.16]<nil> : Syntax
#[lambda-case-test.kl:9.16-9.20]<cons> : Syntax
